pkg/tests/tasks/migration/migration_test.go

```diff
@@ -0,0 +1,149 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migration
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/app"
+	"github.com/maistra/maistra-test-tool/pkg/tests/ossm"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/curl"
+	"github.com/maistra/maistra-test-tool/pkg/util/ns"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/version"
+)
+
+func TestMigrationSimpleClusterWide(t *testing.T) {
+	test.NewTest(t).MinVersion(version.SMCP_2_6).Groups(test.Full).Run(func(t test.TestHelper) {
+		t.Cleanup(func() {
+			oc.DeleteResource(t, "", "Istio", "example")
+			oc.DeleteResource(t, "", "IstioCNI", "default")
+		})
+		defaults := ossm.DefaultClusterWideSMCP()
+		ossm.BasicSetup(t)
+		templ := `apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: {{ .Name }}
+spec:
+  version: {{ .Version }}
+  tracing:
+    type: None
+  security:
+    manageNetworkPolicy: false
+  gateways:
+    enabled: false
+  policy:
+    type: Istiod
+  addons:
+    grafana:
+      enabled: false
+    kiali:
+      enabled: false
+    prometheus:
+      enabled: false
+  mode: ClusterWide`
+		oc.ApplyTemplate(t, defaults.Namespace, templ, defaults)
+
+		oc.WaitSMCPReady(t, defaults.Namespace, defaults.Name)
+		oc.WaitSMMRReady(t, defaults.Namespace)
+		oc.Label(t, "", "Namespace", ns.Bookinfo, "istio-injection=enabled")
+		t.LogStep("Update SMMR to include bookinfo namespace")
+		// Wait for SMMR to include bookinfo
+		oc.DefaultOC.WaitFor(t, defaults.Namespace, "ServiceMeshMemberRoll", "default", `jsonpath='{.status.configuredMembers[?(@=="bookinfo")]}'`)
+
+		t.LogStep("Create 3.0 controlplane and IstioCNI")
+		istio := `apiVersion: sailoperator.io/v1alpha1
+kind: Istio
+metadata:
+  name: example
+spec:
+  namespace: istio-system
+  version: v1.24.1`
+		oc.ApplyString(t, "", istio)
+		oc.DefaultOC.WaitFor(t, "", "Istio", "example", "condition=Ready")
+		oc.CreateNamespace(t, "istio-cni")
+		istioCNI := `apiVersion: sailoperator.io/v1alpha1
+kind: IstioCNI
+metadata:
+  name: default
+spec:
+  namespace: istio-cni
+  version: v1.24.1`
+		oc.ApplyString(t, "", istioCNI)
+
+		t.LogStep("Install bookinfo and bookinfo gateway")
+		app.InstallAndWaitReady(t, app.Bookinfo(ns.Bookinfo))
+
+		oc.ApplyFile(t, ns.Bookinfo, "bookinfo-gateway.yaml")
+		oc.DefaultOC.WaitDeploymentRolloutComplete(t, "bookinfo", "bookinfo-gateway")
+		ip := oc.DefaultOC.Invokef(t, "kubectl get service -n %s %s -o jsonpath='{.status.loadBalancer.ingress[0].ip}'", ns.Bookinfo, "bookinfo-gateway")
```

Comment:
I remember there was some problem here. It takes a lot of time on OCP clusters on RHOS to get the IP and AWS is providing hostname instead of IP IIRC? @mkralik3 do you have some example how this should be used in MTT?

Reply:
in MTT, there is only one occurs here: https://github.com/maistra/maistra-test-tool/blob/main/pkg/tests/ossm-federation/federation_common.go#L144

Reply:
I haven't tested this on RHOS or AWS but I added a wait for the service's ingress to have some status and I made it so that we first try to get the IP from the ingress status and if that's empty then we use the hostname so theoretically it should work for either one even if it's taking a long time now.